account_payment_term_extension/models/account_payment_term_holiday.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2013-2016 Camptocamp SA (Yannick Vaucher)
+# Copyright 2004-2016 Odoo S.A. (www.odoo.com)
+# Copyright 2015-2016 Akretion
+# (Alexis de Lattre <alexis.delattre@akretion.com>)
+# Copyright 2018 Simone Rubino - Agile Business Group
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AccountPaymentTermHoliday(models.Model):
+    _name = "account.payment.term.holiday"
+    _description = "Payment Term Holidays"
+
+    payment_id = fields.Many2one(comodel_name="account.payment.term")
+    holiday = fields.Date(required=True)
+    date_postponed = fields.Date(string="Postponed date", required=True)
+
+    @api.constrains("holiday", "date_postponed")
+    def check_holiday(self):
+        for record in self:
+            if fields.Date.from_string(
+                record.date_postponed
+            ) <= fields.Date.from_string(record.holiday):
```

Comment:
Now, this cast is not necessary because `date` fields are the native `date type` of Python.

```suggestion
            if record.date_postponed <= record.holiday:
```

Reply:
Done.